index.js

```diff
@@ -1,8 +1,24 @@
 'use strict';
-module.exports = function (str) {
-	if (typeof str !== 'string') {
-		throw new TypeError('Expected a string');
+
+var replaceBuf = function (buf) {
+	var crlf = '\r\n';
+	var i = buf.indexOf(crlf);
+
+	if (i === -1) {
+		return buf;
+	}
+
+	var start = buf.slice(0, i);
+	var end = replaceBuf(buf.slice(i + crlf.length));
+	var len = i + start.length + end.length;
+
+	return Buffer.concat([start, '\n', end], len);
+};
+
+module.exports = function (x) {
+	if (typeof x !== 'string' && !Buffer.isBuffer(x)) {
+		throw new TypeError('Expected a string or a buffer');
 	}
 
-	return str.replace(/\r\n/g, '\n');
+	return Buffer.isBuffer(x) ? replaceBuf(x) : x.replace(/\r\n/g, '\n');
```

Comment:
Maybe add `var CRLF = '\r\n';` at the top and reuse it here with `x.replace(new RegExp(CRLF, 'g'), '\n')` and in the `replaceBuf` function?